src/query/nested_query.rs

```diff
@@ -0,0 +1,912 @@
+        self.child_query.as_ref()
+    }
+    pub fn score_mode(&self) -> ScoreMode {
+        self.score_mode
+    }
+    pub fn ignore_unmapped(&self) -> bool {
+        self.ignore_unmapped
+    }
+}
+
+impl fmt::Debug for NestedQuery {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("NestedQuery")
+            .field("path", &self.path)
+            .field("score_mode", &self.score_mode)
+            .field("ignore_unmapped", &self.ignore_unmapped)
+            .finish()
+    }
+}
+
+impl QueryClone for NestedQuery {
+    fn box_clone(&self) -> Box<dyn Query> {
+        Box::new(NestedQuery {
+            path: self.path.clone(),
+            child_query: self.child_query.box_clone(),
+            score_mode: self.score_mode,
+            ignore_unmapped: self.ignore_unmapped,
+        })
+    }
+}
+
+impl Query for NestedQuery {
+    fn weight(&self, enable_scoring: EnableScoring<'_>) -> crate::Result<Box<dyn Weight>> {
+        let mut path_builder = JsonPathWriter::new();
+        for seg in &self.path {
+            path_builder.push(seg);
+        }
+
+        let path_str = path_builder.as_str();
+        let parent_flag_name = format!("_is_parent_{}", path_str);
+
+        let schema = enable_scoring.schema();
+        let parent_field = match schema.get_field(&parent_flag_name) {
+            Ok(f) => f,
+            Err(_) if self.ignore_unmapped => {
+                return Ok(Box::new(NoMatchWeight));
+            }
+            Err(_) => {
+                return Err(TantivyError::SchemaError(format!(
+                    "NestedQuery path '{}' not mapped, and ignore_unmapped=false",
+                    self.path.join(".")
+                )));
+            }
+        };
+
+        let exclude_parent_term = Term::from_field_bool(parent_field, true);
+        let exclude_parent_q =
+            TermQuery::new(exclude_parent_term.clone(), IndexRecordOption::Basic);
+
+        let child_plus_exclude = BooleanQuery::new(vec![
+            (Occur::Must, self.child_query.box_clone()),
+            (Occur::MustNot, Box::new(exclude_parent_q)),
+        ]);
+
+        let block_join_query = ToParentBlockJoinQuery::new(
+            Box::new(child_plus_exclude),
+            Arc::new(NestedParentBitSetProducer::new(parent_field)),
+            self.score_mode,
+        );
+
+        let weight = block_join_query.weight(enable_scoring)?;
+        Ok(weight)
+    }
+
+    fn explain(&self, searcher: &Searcher, doc_address: DocAddress) -> crate::Result<Explanation> {
+        let w = self.weight(EnableScoring::enabled_from_searcher(searcher))?;
+        let explanation = w.explain(
+            searcher.segment_reader(doc_address.segment_ord),
+            doc_address.doc_id,
+        )?;
+        Ok(explanation)
+    }
+
+    fn count(&self, searcher: &Searcher) -> crate::Result<usize> {
+        let w = self.weight(EnableScoring::disabled_from_searcher(searcher))?;
```

Comment:
let's avoid one letter variable like this.
```suggestion
        let w = self.weight(EnableScoring::disabled_from_searcher(searcher))?;
```